Source/WebKit/WebProcess/Network/WebResourceLoader.cpp

```diff
@@ -115,39 +117,43 @@ MainFrameMainResource WebResourceLoader::mainFrameMainResource() const
 void WebResourceLoader::willSendRequest(ResourceRequest&& proposedRequest, IPC::FormDataReference&& proposedRequestBody, ResourceResponse&& redirectResponse, CompletionHandler<void(ResourceRequest&&, bool)>&& completionHandler)
 {
     Ref<WebResourceLoader> protectedThis(*this);
+    RefPtr coreLoader = m_coreLoader;
 
     // Make the request whole again as we do not normally encode the request's body when sending it over IPC, for performance reasons.
     proposedRequest.setHTTPBody(proposedRequestBody.takeData());
 
     LOG(Network, "(WebProcess) WebResourceLoader::willSendRequest to '%s'", proposedRequest.url().string().latin1().data());
     WEBRESOURCELOADER_RELEASE_LOG(WEBRESOURCELOADER_WILLSENDREQUEST);
     
-    if (RefPtr frame = m_coreLoader->frame()) {
+    if (RefPtr frame = coreLoader->frame()) {
         if (RefPtr page = frame->page()) {
             if (!page->allowsLoadFromURL(proposedRequest.url(), mainFrameMainResource()))
                 proposedRequest = { };
         }
     }
 
-    m_coreLoader->willSendRequest(WTFMove(proposedRequest), redirectResponse, [this, protectedThis = WTFMove(protectedThis), completionHandler = WTFMove(completionHandler)] (ResourceRequest&& request) mutable {
-        if (!m_coreLoader || !m_coreLoader->identifier()) {
+    coreLoader->willSendRequest(WTFMove(proposedRequest), redirectResponse, [this, protectedThis = WTFMove(protectedThis), completionHandler = WTFMove(completionHandler)] (ResourceRequest&& request) mutable {
+        RefPtr coreLoader = m_coreLoader;
+        if (!coreLoader || !coreLoader->identifier()) {
             WEBRESOURCELOADER_RELEASE_LOG(WEBRESOURCELOADER_WILLSENDREQUEST_NO_CORELOADER);
             return completionHandler({ }, false);
         }
 
         WEBRESOURCELOADER_RELEASE_LOG(WEBRESOURCELOADER_WILLSENDREQUEST_CONTINUE);
-        completionHandler(WTFMove(request), m_coreLoader->isAllowedToAskUserForCredentials());
+        completionHandler(WTFMove(request), coreLoader->isAllowedToAskUserForCredentials());
     });
 }
 
 void WebResourceLoader::didSendData(uint64_t bytesSent, uint64_t totalBytesToBeSent)
 {
-    m_coreLoader->didSendData(bytesSent, totalBytesToBeSent);
+    RefPtr coreLoader = m_coreLoader;
+    coreLoader->didSendData(bytesSent, totalBytesToBeSent);
```

Comment:
Ditto. We could do:
```cpp
RefPtr { m_coreLoader }->didSendData(bytesSent, totalBytesToBeSent)
```

But in this case, we could add a new helper like `protectedCoreLoader()` which returns RefPtr and simply use that:
```cpp
    protectedCoreLoader()->didSendData(bytesSent, totalBytesToBeSent);
```